scripts/test/integration-setup/index.js

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+
+'use strict';
+
+require('essentials');
+require('log-node')();
+
+const log = require('log').get('serverless:scripts');
+const awsRequest = require('@serverless/test/aws-request');
+const fsPromises = require('fs').promises;
+const path = require('path');
+const { SHARED_INFRA_TESTS_CLOUDFORMATION_STACK } = require('../../../test/utils/cludformation');
+
+(async () => {
+  log.notice('Starting setup of integration infrastructure...');
+
+  const [cfnTemplate, kafkaServerProperties] = await Promise.all([
+    fsPromises.readFile(path.join(__dirname, 'cloudformation.yml'), 'utf8'),
+    fsPromises.readFile(path.join(__dirname, 'kafka.server.properties')),
+  ]);
+
+  log.notice('Checking if integration tests CloudFormation stack already exists...');
+  try {
+    await awsRequest('CloudFormation', 'describeStacks', {
+      StackName: SHARED_INFRA_TESTS_CLOUDFORMATION_STACK,
+    });
+    log.error('Integration tests CloudFormation stack already exists. Quitting.');
+    return;
```

Comment:
I'm wondering, that it'll probably be good to proceed with _update_ operation if stack is there.

I can imagine that at some point we may want to extend this stack, and currently this script won't handle that well (we will have to remove the stack and re-create it, which might in many cases be quite time taking).

What do you think?

Reply:
That makes sense - I believe in that case I would like to have it as two separate branches of logic as when we want to perform an update, we will need to skip some of the operations (e.g. creation of MSK Cluster Configuration). Note: We might still need to go through the whole process if there are dependencies that are not set up via CloudFormation.

Reply:
I've adjusted the setup script so now it'll check if the stack can be updated and will try to perform an update :+1: